lint/appendAssign_checker.go

```diff
@@ -0,0 +1,100 @@
+package lint
+
+//! Detects suspicious append result assignments.
+//
+// @Before:
+// p.positives = append(p.negatives, x)
+// p.negatives = append(p.negatives, y)
+//
+// @After:
+// p.positives = append(p.positives, x)
+// p.negatives = append(p.negatives, y)
+
+import (
+	"go/ast"
+	"go/token"
+	"go/types"
+
+	"github.com/go-toolsmith/astequal"
+	"github.com/go-toolsmith/astp"
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+func init() {
+	addChecker(&appendAssignChecker{}, attrExperimental)
+}
+
+type appendAssignChecker struct {
+	checkerBase
+}
+
+func (c *appendAssignChecker) VisitStmt(stmt ast.Stmt) {
+	assign, ok := stmt.(*ast.AssignStmt)
+	if !ok || assign.Tok != token.ASSIGN || len(assign.Lhs) != len(assign.Rhs) {
+		return
+	}
+	for i, rhs := range assign.Rhs {
+		call, ok := rhs.(*ast.CallExpr)
+		if !ok || qualifiedName(call.Fun) != "append" {
+			continue
+		}
+		c.checkAppend(assign.Lhs[i], call)
+	}
+}
+
+func (c *appendAssignChecker) matchSlices(cause ast.Node, x, y ast.Expr) {
```

Comment:
IMO, can we move this method after `checkAppend`? :3

Reply:
Done.